docs/source/reference/ha.rst

```diff
@@ -0,0 +1,301 @@
+distribute work or take over new work if some nodes in the cluster disappear. It would however be possible to a
+sensor itself to be implemented with HA in mind so that the same sensor can be deployed on multiple nodes with
+the sensor managing active-active or active-passive. Providing some platform level HA support for sensors is
+likely to be an enhancement to StackStorm in future releases.
+
+
+st2rulesengine
+^^^^^^^^^^^^^^
+This is a dual purpose process - its main function is to evaluate rules when it sees TriggerInstances and
+decide if an ActionExecution is to be requested. It needs access to MongoDB to locate rules and RabbitMQ
+to listen for TriggerInstances and request ActionExecutions. The auxiliary purpose of this process is to
+run all the defined timers. See `Timers <ref-rule-timers>` for specifics on setting up timers via rules.
+
+n st2rulesengine can run in active-active with only connections to MongoDB and RabbitMQ. All these will share the
+TriggerInstance load and naturally pick up more work if one or more of the process becomes unavailable. The timer
+function in each of the RulesEngine is not HA compatible. In the interim it is possible to disable the timer
+in all but one of the st2rulesengine to avoid duplicate timer events, expect this to be fixed in a future |st2| release.
+
+st2actionrunner
+^^^^^^^^^^^^^^^
+All ActionExecutions are handled for execution by st2actionrunner. It manages the full life-cycle of an execution from
+scheduling to one of the terminal states.
+
+n st2actionrunner can run in active-active with only connections to MongoDB and RabbitMQ. Work gets naturally
+distributed across runners via RabbitMQ. Adding more st2actionrunner increases the ability of |st2| to execute actions.
+
+In a proper distributed setup it is recommended to setup Zookeeper or Redis to provide a distributed co-ordination
+layer. See :doc:`Policies </policies>`. Using a default file based co-ordination backend will not work as it would
+in a single box deployment.
+
+st2resultstracker
+^^^^^^^^^^^^^^^^^
+Tracks results of execution handed over to mistral. It requires access to MongoDB and RabbitMQ to perform its function.
+
+n st2resultstracker will co-operate with each other to perform work. At startup though there is a possibility
+of work however there will not be any negative consequences to this duplicate work. Specifically the jobs to track results
+also get stored in the DB in case there is no worker to take over the work and this pattern makes all result trackers
+pick up the same initial work set. Once this work set is exhausted all subsequents tasks are round-robined.
+
+st2notifier
+^^^^^^^^^^^
+This is a dual purpose process - its main function is to generate ``st2.core.actiontrigger`` and ``st2.core.notifytrigger``
+based on the completion of ActionExecution. The auxiliary purpose is to act as a backup scheduler for actions that may
```

Comment:
It is https://github.com/StackStorm/st2/blob/master/st2actions/st2actions/notifier/scheduler.py

The scheduler recovers any delayed executions. It is a safety-valve in case execution do not get scheduled once they have been delayed due to some policy https://docs.stackstorm.com/policies.html